integration-test-remote/src/test/java/io/cdap/cdap/app/etl/google/drive/GoogleDriveTest.java

```diff
@@ -0,0 +1,570 @@
+        String content = getFileContent(file.getId());
+        switch (fileName) {
+          case TEST_TEXT_FILE_NAME:
+            assertEquals(TEST_TEXT_FILE_CONTENT, content);
+            break;
+          case TEST_DOC_FILE_NAME:
+            // check BOM
+            assertEquals('\uFEFF', content.charAt(0));
+            assertEquals(TEST_DOC_FILE_CONTENT, content.replace("\uFEFF", ""));
+            break;
+          case TEST_SHEET_FILE_NAME:
+            assertEquals(TEST_SHEET_FILE_CONTENT, content);
+            break;
+          default:
+            fail(String.format("Invalid file name after pipeline completion: '%s', content: '%s'", fileName, content));
+        }
+      } catch (IOException e) {
+        fail(String.format("Exception during test results check: [%s]", e.getMessage()));
+      }
+    });
+  }
+
+  @Test
+  public void testAllFileTypesNamedAndMimed() throws Exception {
+    Map<String, String> sourceProps = getSourceMinimalDefaultConfigs();
+    sourceProps.put("fileTypesToPull", "binary,documents,spreadsheets");
+    sourceProps.put("fileMetadataProperties", "name,mimeType");
+    Map<String, String> sinkProps = getSinkMinimalDefaultConfigs();
+    sinkProps.put("schemaNameFieldName", "name");
+    sinkProps.put("schemaMimeFieldName", "mimeType");
+
+    DeploymentDetails deploymentDetails =
+      deployApplication(sourceProps, sinkProps,
+                        GOOGLE_DRIVE_PLUGIN_NAME + "-testAllFileTypesNamedAndMimed");
+    startWorkFlow(deploymentDetails.getAppManager(), ProgramRunStatus.COMPLETED);
+
+    // check number of rows in and out
+    checkRowsNumber(deploymentDetails, 3);
+
+    List<File> destFiles = getFiles(sinkFolderId);
+    assertEquals(3, destFiles.size());
+
+    destFiles.forEach(file -> {
+      Assert.assertNotNull(file.getName());
+      try {
+        String fileName = file.getName();
+        String mimeType = file.getMimeType();
+        String content = getFileContent(file.getId());
+        switch (fileName) {
+          case TEST_TEXT_FILE_NAME:
+            assertEquals(TEST_TEXT_FILE_CONTENT, content);
+            assertEquals(TEXT_PLAIN_MIME, mimeType);
+            break;
+          case TEST_DOC_FILE_NAME:
+            // check BOM
+            assertEquals('\uFEFF', content.charAt(0));
+            assertEquals(TEST_DOC_FILE_CONTENT, content.replace("\uFEFF", ""));
+            assertEquals(TEXT_PLAIN_MIME, mimeType);
+            break;
+          case TEST_SHEET_FILE_NAME:
+            assertEquals(TEST_SHEET_FILE_CONTENT, content);
+            assertEquals(TEXT_CSV_MIME, mimeType);
+            break;
+          default:
+            fail(String.format("Invalid file name after pipeline completion: '%s', content: '%s', mime type: '%s'",
+                               fileName, content, mimeType));
+        }
+      } catch (IOException e) {
+        fail(String.format("Exception during test results check: [%s]", e.getMessage()));
```

Comment:
Can we add info here for what file is exception and some other details possibly.

Reply:
done